README.md

```diff
@@ -7,59 +7,411 @@
+  }
+})
 ```
 
-# Generate a trustline prefix
+## Example Code with Unconditional Payment-Based Settlement
+
+Unconditional payment settlement secures a trustline balance by sending payments
+on a system that doesn't support conditional transfers. Hashed timelock transfers
+go through plugin virtual like a clearing layer, and every so often a settlement
+is sent to make sure the amount secured on the ledger doesn't get too far from
+the finalized amount owed.
 
-If you want to generate a trustline prefix from a secret key, peer's public key, and currency,
-you can access the method through the token class:
+Unlike creating a claim, sending a payment has side-effects (it alters an
+external system). Therefore, the code is slightly more complicated.
 
 ```js
-const Token = require('ilp-plugin-virtual/src/util/token')
-const prefix = Token.prefix({
-  secretKey: 'o9Lt0oZFek1ArM_A0HUAq8M8edRuoSeLjX8i10sVUiY',
-  peerPublicKey: 'KRixgcBCBdyQln7IBYiopjuNO78QSFtXgOwP1sbsCSk',
-  currency: 'USD'
-})
+const { MakePluginVirtual } = require('ilp-plugin-virtual')
+const { NotAcceptedError } = require('ilp-plugin-shared').Errors
+const Network = require('some-example-network')
+const BigNumber = require('bignumber.js')
+
+return MakePluginVirtual({
+  connect: async function (ctx, opts) {
+    await Network.connectToNetwork()
+
+    // In this type of payment channel backend, we create a log of incoming
+    // settlements to track all the transfers sent to us on the ledger we're
+    // using for settlement.  We use a transferLog in order to make sure a
+    // single transfer can't be added twice.
+    ctx.state.incomingSettlements = ctx.backend.getTransferLog('incoming_settlements')
+
+    // The amount settled is used to track how much we've paid out in total.
+    // We'll go deeper into how it's used in the `createOutgoingClaim`
+    // function.
+    ctx.state.amountSettled = ctx.backend.getMaxValueTracker('amount_settled')
+
+    // In this type of payment channel backend, the unsecured balance we want
+    // to limit is the total amount of incoming transfers minus the sum of all
+    // the settlement transfers we've received.
+    ctx.state.maxUnsecured = opts.maxUnsecured
+  },
+
+  handleIncomingPrepare: async function (ctx, transfer) {
+    const incoming = await ctx.transferLog.getIncomingFulfilledAndPrepared() 
+
+    // Instead of getting the best claim, we're getting the sum of all our
+    // incoming settlement transfers. This tells us how much incoming money has
+    // been secured.
+    const amountReceived = await ctx.state.incomingSettlements.getIncomingFulfilledAndPrepared()
+
+    // The peer can default on 'incoming - amountReceived', so we want to limit
+    // that amount.
+    const exceeds = new BigNumber(incoming)
+      .subtract(amountReceived)
+      .greaterThan(ctx.state.maxUnsecured)
+
+    if (exceeds) {
+      throw new NotAcceptedError(transfer.id + ' exceeds max unsecured balance')
+    }
+  },
```

Comment:
If this function doesn't throw does it automatically accept the incoming prepared transfer?